src/ctapipe/tools/calculate_pixel_stats.py

```diff
@@ -100,18 +100,20 @@ def setup(self):
         self.input_data.dl1_images = True
         # Load the subarray description from the input file
         subarray = SubarrayDescription.from_hdf(self.input_data.input_url)
-        # Get the telescope ids from the input data or use the allowed_tels configuration
-        self.tel_ids = (
-            subarray.tel_ids if self.allowed_tels is None else self.allowed_tels
+        # Select a new subarray if the allowed_tels configuration is used
+        self.subarray = (
+            subarray
+            if self.allowed_tels is None
+            else subarray.select_subarray(self.allowed_tels, "Subarray")
```

Comment:
I'd leave the default name here, it will create a somewhat nice description of which telescopes were selected from which subarray.